pulsar-client-cpp/lib/Commands.cc

```diff
@@ -66,7 +68,9 @@ SharedBuffer Commands::newConsumerStats(BaseCommand& cmd, uint64_t consumerId, u
     CommandConsumerStats* consumerStats = cmd.mutable_consumerstats();
     consumerStats->set_consumer_id(consumerId);
     consumerStats->set_request_id(requestId);
-    return writeMessageWithSize(cmd);
+    const SharedBuffer& buffer = writeMessageWithSize(cmd);
+    cmd.release_consumerstats();
```

Comment:
as mentioned into other comment. I kept it because caller is passing `BaseCommand` and in future if we have any conditional field update then we have the same issue.